cirq/linalg/decompositions.py

```diff
@@ -667,6 +796,13 @@ def kak_decomposition(
         'An Introduction to Cartan's KAK Decomposition for QC Programmers'
         https://arxiv.org/abs/quant-ph/0507171
     """
+    if isinstance(unitary_object, KakDecomposition):
+        return unitary_object
+    if isinstance(unitary_object, np.ndarray):
+        mat = unitary_object
```

Comment:
This special-casing to make numpy arrays work in a context that expects a unitary seems potentially useful elsewhere. Have you considered moving this to the unitary protocol?

Reply:
I have considered it. One thing that worries me about it is what to do if the array is not unitary. Do we need to check?

Reply:
Currently, the protocol raises an exception when the argument fails to satisfy our requirements, so I'd do the same in the new case.

Minor caveat: Existing cases justify its use of the `TypeError` exception. The present case fits `ValueError` better.